### ISSUE TITLE
Vulkan development headers minimum version

### ISSUE THREAD

**Issue by gunqqer** [catsout/wallpaper-engine-kde-plugin#185]
**Describe the bug**<br/>
As of vulkan version 1.2.182 Vulkan-Hpp headers were split into multiple files, which this repo assumed has been done. Ubuntu 20.04 has v1.2.131.2-1 so after commit 8a2563dddaaa8a60c4d5af39e57397d7ef836b99 this plugin no longer is able to be compiled. It should be added to the README that this version is needed and Ubuntu 20.04 doesn't have it. 

**Workshop link**<br/>
NA

**System Information**
- Kde: 5.24.5
- Qt: 5.15.4
- OS: Ubuntu 20.04 LTS
- Installation Source: git

**Additional context**<br/>
NA

**Comment by catsout** [comment]
I known the problem that old distribution don't have new version of vulkan hpp, so I don't use the split vulkan hpp.  
I introduce `SPIRV-Reflect` in this commit.  
Please give the log of `make`.

**Comment by gunqqer** [comment]
```In file included from /tmp/wallpaper-engine-kde-plugin/src/backend_scene/src/Vulkan/Instance.cpp:225:
/tmp/wallpaper-engine-kde-plugin/src/backend_scene/src/Vulkan/Shader_impl.hpp:16:10: fatal error: vulkan/vulkan_enums.hpp: No such file or directory
   16 | #include <vulkan/vulkan_enums.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~
```
The issue is Shader_impl.hpp trying to include vulkan_enums.hpp
I can provide the full build log but the only other thing of note is strict-aliasing warnings.

**Comment by catsout** [comment]
Fixed.  
This is added by the nvim lsp autocompletion plugin. I didn't notice it.

**Comment by gunqqer** [comment]
Nice! Fixed by eca8c2b361d1b876d57bac6c5eea5f09aaf4a806
Closing issue
